File: week01/Theme02/Theme02_Code01.cpp
// Theme02_Code01.cpp

#include <iostream>
#include <queue>
using namespace std;

ostream& operator <<(ostream& os, queue<int> q) {
	while (!q.empty()) {
		os << q.front() << " ";
		q.pop();
	}
	return os;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	queue<int> q;

	q.push(0);
	q.push(1);
	q.push(2);

	cout << q << endl; // 0 1 2
	
	q.push(3);
	q.push(4);
	q.pop();
	q.pop();
	q.pop();

	cout << q << endl; // 2 3 4
}

File: week01/Theme02/Theme02_Code01.py
# Theme01_Code01.py

import queue

def print_queue(q):
    temp = queue.Queue()
    while not q.empty():
        item = q.get()
        print(item, end=" ")
        temp.put(item)
    while not temp.empty():
        q.put(temp.get())
    print()

q = queue.Queue()

q.put(0)
q.put(1)
q.put(2)

print_queue(q) # 0 1 2

q.put(3)
q.put(4)
print(q.get()) # 0
print(q.get()) # 1

print_queue(q) # 2 3 4